### PR TITLE
Merge crafting_allowed and crafting_can_see into one function

### PR DESCRIPTION
**Pull request by JornVernee**
This is mostly the make the checks if crafting is allowed cleaner in the code, because we only call 1 function. This has the added benefit that we can now open the crafting menu even if our morale is too low (we still can't craft). As was intended: https://github.com/CleverRaven/Cataclysm-DDA/issues/14028

As you can see in the diff; `crafting_allowed` is called before the crafting menu is opened (`select_crafting_recipe`). `crafting_allowed` checks if we have enough morale to craft. If the call is before the opening of the crafting menu, we can't open the menu if we don't enough morale. (I also tested this by putting the call to `crafting_can_see` in the same place, I could indeed not open the menu when it was too dark)

### CODE REVIEW

**Comment by JornVernee** [comment]
As far as I understand it, the lua bindings are used to call functions/look up fields from the debug menu. Or are they also used for other things? the documentation on lua (LUA_SUPPORT.md) explains how to add functions ,wrap classes, etc. But not what might actually be done with lua.

This makes it hard for me to determine whether bindings are safe to remove.

**Comment by kevingranade** [comment]
If nothing else you should be able to just search through the tiny amount
of Lua included in the repository for the method names or attributes in
question.
That having been said, that cleanup isn't necessarily something you have to
take care of.

**Comment by JornVernee** [comment]
I thought someone might have some lua files not in the repo that reference it (like a mod).

**Comment by kevingranade** [comment]
They might, but if they don't care to push to the repo, I don't care to try
and keep their code working.

File: src/crafting.cpp [hunk 1]
```diff
@@ -307,11 +307,7 @@ bool player::crafting_allowed()
         add_msg(m_info, _("Your morale is too low to craft..."));
         return false;
     }
-    return true;
-}
```
Comment (kevingranade) [inline src/crafting.cpp]: hah, convenient that you can just get rid of the separators.

**Comment by JornVernee** [comment]
I added the indicator discussed in the issue as well, I felt like this PR wasn't complete without it.

**Pull request merged**